hydroflow/tests/surface_async.rs

```diff
@@ -258,8 +255,113 @@ pub async fn test_futures_stream_sink() -> Result<(), Box<dyn Error>> {
         _ = tokio::time::sleep(Duration::from_secs(1)) => (),
     };
 
-    let seen: Vec<_> = hydroflow::util::collect_ready(seen_recv).await;
+    let seen: Vec<_> = collect_ready(seen_recv);
     rassert_eq!(&std::array::from_fn::<_, MAX, _>(|i| i), &*seen)?;
 
     Ok(())
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn asynctest_dest_sink_bounded_channel() {
+    // In this example we use an _bounded_ channel for our `Sink`. This is for demonstration only,
+    // instead you should use [`hydroflow::util::unbounded_channel`]. A bounded channel results in
+    // `Hydroflow` buffering items internally instead of within the channel.
+    let (send, recv) = tokio::sync::mpsc::channel::<usize>(5);
+    let send = tokio_util::sync::PollSender::new(send);
+    let mut recv = tokio_stream::wrappers::ReceiverStream::new(recv);
+
+    let mut flow = hydroflow_syntax! {
+        source_iter(0..10) -> dest_sink(send);
+    };
+    tokio::time::timeout(std::time::Duration::from_secs(1), flow.run_async())
+        .await
+        .expect_err("Expected time out");
+
+    // Only 5 elemts received due to buffer size
+    let out: Vec<_> = collect_ready(&mut recv);
+    assert_eq!(&[0, 1, 2, 3, 4], &*out);
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn asynctest_dest_sink_duplex() {
```

Comment:
this is complicated and I'm not sure if you intend for users to do this?

Reply:
not necessarily intended for users, but works as a test